build-tools/packages/build-tools/src/test/typeTestGenerator.test.ts

```diff
@@ -0,0 +1,243 @@
+			assert.strictEqual(result, expectedPath);
+		});
+	});
+
+	describe("Test getTypeRollupPathFromExtractorConfig", () => {
+		// Create temp directory for testing
+		const nodeModulesDir = path.join(__dirname, "node_modules");
+		const previousBasePath = path.join(nodeModulesDir, previousPackageName);
+		let extractorConfigOptions: IExtractorConfigPrepareOptions;
+
+		before(() => {
+			fs.mkdirSync(nodeModulesDir);
+			fs.mkdirSync(previousBasePath);
+		});
+
+		after(() => {
+			fs.rmSync(nodeModulesDir, { recursive: true });
+		});
+
+		it("Should return undefined if API Extractor config is not found", () => {
+			const result = utils.getTypeRollupPathFromExtractorConfig(
+				"alpha",
+				extractorConfigOptions,
+			);
+			assert.strictEqual(result, undefined);
+		});
+
+		it("Should return undefined if dtsRollup config is not found", () => {
+			fs.writeFileSync(path.join(previousBasePath, "api-extractor.json"), "{}");
+			const result = utils.getTypeRollupPathFromExtractorConfig(
+				"alpha",
+				extractorConfigOptions,
+			);
+			assert.strictEqual(result, undefined);
+			fs.rmSync(path.join(previousBasePath, "api-extractor.json"));
+		});
+
+		it("Should return undefined if rollup path for the specified type is not found", () => {
+			extractorConfigOptions = {
+				configObject: {
+					mainEntryPointFilePath: "",
+					dtsRollup: {
+						enabled: true,
+						untrimmedFilePath: "untrimmed.d.ts",
+					},
+				},
+				configObjectFullPath: "",
+				packageJsonFullPath: "",
+			};
+			const result = utils.getTypeRollupPathFromExtractorConfig(
+				"alpha",
+				extractorConfigOptions,
+			);
+			assert.strictEqual(result, undefined);
+		});
+
+		it("Should return the rollup path for the specified type", () => {
+			extractorConfigOptions = {
+				configObject: {
+					mainEntryPointFilePath: "",
+					dtsRollup: {
+						enabled: true,
+						alphaTrimmedFilePath: "alpha.d.ts",
+					},
+				},
+				configObjectFullPath: "",
+				packageJsonFullPath: "",
+			};
+			const result = utils.getTypeRollupPathFromExtractorConfig(
+				"alpha",
+				extractorConfigOptions,
+			);
+			assert.strictEqual(result, "alpha.d.ts");
+		});
+	});
+
+	describe("Test getTypePathFromExport", () => {
+		const nodeModulesDir = path.join(__dirname, "node_modules");
+		const previousBasePath = path.join(nodeModulesDir, previousPackageName);
+
+		after(() => {
+			fs.rmSync(nodeModulesDir, { recursive: true });
+		});
+
+		it("should return undefined if exports field is missing", () => {
+			const result = utils.getTypePathFromExport(packageObject, previousBasePath);
+			assert.strictEqual(result, undefined);
+		});
+
+		it("should throw an error if both import and require resolutions are missing", () => {
+			packageObject.exports = { ".": {} };
+
+			assert.throws(() => {
+				utils.getTypePathFromExport(packageObject, previousBasePath);
+			}, "Type definition file path could not be determined");
```

Comment:
Note: this doesn't do exactly what you would expect. This will assert that an error is thrown, and if not, the test will fail with an error message that says "Type definition file path could not be determined".

To assert on the error message, check out the pattern used here:
https://github.com/microsoft/FluidFramework/blob/main/packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts#L127

Reply:
That specific example uses our validate function which is only needed because we transform asserts in production code to a number, so standard mechanisms of checking exception properties don't work.

But here, we can just use assert's capability directly, e.g.

```ts
			// Using regular expressions to validate error message:
			assert.throws(
				() => utils.getTypePathFromExport(packageObject, previousBasePath),
				/Type definition file path could not be determined from the 'exports' field using the default export entry '.'/,
			);
```

Reply:
Node docs: https://nodejs.org/docs/latest-v18.x/api/assert.html#assertthrowsfn-error-message

There's another example there, plus other examples of the types of deep property equality checks you can do.